internal/namespace/typesystem.go

```diff
@@ -474,6 +475,97 @@ func (nts *TypeSystem) typeSystemForNamespace(ctx context.Context, namespaceName
 	return BuildNamespaceTypeSystem(nsDef, nts.lookupNamespace)
 }
 
+// Returns the relation names (right-hand side) of an arrow expression given its left hand-side value
+//
+// Sample:
+//
+// ```
+// definition resource {
+//     relation reader: user
+//     relation writer: user
+//     relation parent: resource_group | resource
+
+//	    permission read = reader + writer + parent->read
+//	    permission write = reader
+//	}
+//
+// ```
+// Calling ResolveArrowRelations('parent') returns ['read']
+func (nts *TypeSystem) ResolveArrowRelations(ttuRelation string) ([]string, error) {
+	var referencedRelations []string
+	for _, relation := range nts.Namespace().Relation {
+		if nts.IsPermission(relation.Name) {
+			found, err := nts.searchReferencedRelationsInUsersetRewrite(ttuRelation, relation.UsersetRewrite)
+			if err != nil {
+				return nil, err
+			}
+			referencedRelations = append(referencedRelations, found...)
+		}
+	}
+	return referencedRelations, nil
+}
+
+func (nts *TypeSystem) searchReferencedRelationsInUsersetRewrite(ttuRelation string, usr *core.UsersetRewrite) ([]string, error) {
+	switch rw := usr.RewriteOperation.(type) {
+	case *core.UsersetRewrite_Union:
+		return nts.searchReferencedRelationsInSetOperation(ttuRelation, rw.Union)
+	case *core.UsersetRewrite_Intersection:
+		return nts.searchReferencedRelationsInSetOperation(ttuRelation, rw.Intersection)
+	case *core.UsersetRewrite_Exclusion:
+		return nts.searchReferencedRelationsInSetOperation(ttuRelation, rw.Exclusion)
+	default:
+		return nil, errors.New("userset rewrite operation not implemented")
+	}
+}
+
+func (nts *TypeSystem) searchReferencedRelationsInSetOperation(ttuRelation string, so *core.SetOperation) ([]string, error) {
+	var foundRelations []string
+	for _, childOneof := range so.Child {
+		switch child := childOneof.ChildType.(type) {
+		case *core.SetOperation_Child_XThis:
+			break
+		case *core.SetOperation_Child_ComputedUserset:
+			found, err := nts.searchReferencedRelationsInComputedUserset(ttuRelation, child.ComputedUserset)
```

Comment:
Just have this branch be empty

Reply:
done (I didn't know that in Go the default behaviour was not to fallthrough, thanks !)